File: remote-control/src/test/unit/java/com/thoughtworks/selenium/grid/remotecontrol/RegistrationRequestTest.java
package com.thoughtworks.selenium.grid.remotecontrol;

import static junit.framework.Assert.assertEquals;
import org.jbehave.classmock.UsingClassMock;
import org.junit.Test;

import java.io.IOException;

/**
 * {@link RegistrationRequest} unit test class.
 */
public class RegistrationRequestTest extends UsingClassMock {

    @Test
    public void postToHubUsingURLProvidedInConstructor() throws IOException {
      final HubRequest request = new RegistrationRequest("http://thehub.url:4444", "", "", "");
        assertEquals("http://thehub.url:4444/registration-manager/register", request.postMethod().getURI().toString());
    }

}

File: remote-control/src/test/unit/com/thoughtworks/selenium/grid/remotecontrol/registration_request_test.rb
require File.expand_path(File.dirname(__FILE__) + "/../../../../../unit_test_helper")

include_class 'com.thoughtworks.selenium.grid.remotecontrol.RegistrationRequest'

class RegistrationRequestTest < Test::Unit::TestCase

  test "Post to Hub using URL provided in constructor for registration request" do
    request = RegistrationRequest.new "http://thehub.url:4444", "", "", ""
    assert_equal "http://thehub.url:4444/registration-manager/register", request.post_method.get_uri.to_string
  end

end